### PR TITLE
feat(vector): Create checksum annotations for existingConfig and extraConfigs

### LINKED ISSUE #62: `rollWorkload` only works when using the configmap template
[`lookup`](https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-lookup-function) could be used to access the contents of existing and extra configmaps

### PR DESCRIPTION
**Pull request by spencergilbert**
Signed-off-by: Spencer Gilbert <spencer.gilbert@datadoghq.com>

Closes #62

### CODE REVIEW

File: charts/vector/templates/daemonset.yaml [hunk 1]
```diff
@@ -17,7 +17,20 @@ spec:
     metadata:
       annotations:
       {{- if .Values.rollWorkload }}
+        {{- if .Values.existingConfig }}
+          {{- range $file, $contents := (lookup "v1" "ConfigMap" (print .Release.Namespace) (print .Values.existingConfig)).data }}
```
Comment (spencergilbert) [inline charts/vector/templates/daemonset.yaml:21]: `lookup` fetches a resource via the kubernetes api, and we range through the `data` key which should contain config files like:

```
data:
  myConfig.yaml: |
....
```

File: charts/vector/templates/daemonset.yaml [hunk 2]
```diff
@@ -17,7 +17,20 @@ spec:
...
+        checksum/{{ $file }}: {{ $contents | sha256sum }}
+          {{- end }}
+        {{- else }}
         checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        {{- end }}
+        {{- if .Values.extraConfigs }}
+          {{- range .Values.extraConfigs }}
```
Comment (spencergilbert) [inline charts/vector/templates/daemonset.yaml:28]: This ranges through the list of extraConfigs and calls the api for each of them.

**Review by spencergilbert — COMMENTED** [review]
The outcome looks something like this: ```yaml
      annotations:
        checksum/config.toml: 2d63c8274091d07e5c2d88b26397f595d14ef52d53f93ddf9037f26a0be4b903
        checksum/sinks.toml: fdcc48843a4d44052a075e9f35d9ec9fe363a46af19e994013cef8761eca5de1
        checksum/extra.sink.toml: d9aec60f80f3cac44f7bd9242a6ba03319f7e8e7f95383aa1f620adb8af62331
```

**Review by jszwedko — APPROVED** [review]
👍 thanks for the explanatory comments.

**Pull request merged**